src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -1035,12 +1035,7 @@ impl Builder<'_> {
             rustflags.arg("-Wrustc::internal");
             // cfg(bootstrap) - remove this check when lint is in bootstrap compiler
             if stage != 0 {
-                // Lint is allow by default so downstream tools don't get a lit
-                // they can do nothing about
```

Comment:
I don't think we should apply unstable solutions to problems we think might arise in the future. We shouldn't worry about it since it's not an issue yet.

Reply:
cc @Urgau as you mentioned this being a problem https://github.com/rust-lang/rust/pull/133545#discussion_r1861614714

Reply:
Actually wait, that is not relevant for this change here.

Reply:
Agreed